### PR TITLE
 Image conversion crash fix

### PR DESCRIPTION
**Pull request by yalcinmelihyasin**
This commit fixes the crash caused by the bug. And also
     - Prints more information when image conversion is failed
     - Renames the function that reads min and max value from Float src
     - It does reading in the conversion that needs it.

b/161683278

### CODE REVIEW

File: core/stream/convert.go [hunk 1]
```diff
@@ -195,9 +188,10 @@ func (m *mapping) conv(count int, min, max float64) error {
 		if d.DataType.Signed {
 			return ftos(count, m.dst, m.src)
 		}
-		return ftou(count, m.dst, m.src, min, max)
+		return ftou(count, m.dst, m.src)
+	default:
+		return fmt.Errorf("Cannot convert from Unknown %v to Unknown %v", s, d)
```
Comment (yalcinmelihyasin) [inline core/stream/convert.go:193]: I guess this should never happen. Maybe we can change it with panic?

**Review by pmuetschard — CHANGES_REQUESTED** [review]
The min/max should be calculated across all channels not channel-by-channel.

When fixing a bug, don't add additional changes in at the same time. Make them separate commits. It's hard to see what the actual bug fix is here due to the additional noisy changes. The changes inside the `ftouMinMax` function should be enough to fix the bug, no?

File: core/stream/convert.go [hunk 2]
```diff
@@ -396,7 +398,7 @@ func writeUintClamped(bs *binary.BitStream, bits uint64, count uint32) {
 	bs.Write(u64.Min(bits, limit), count)
 }
 
-func ftouMinMax(count int, dst, src buf, min, max *float64) error {
+func readMinMax(count int, src buf, min, max *float64) error {
```
Comment (pmuetschard) [inline core/stream/convert.go:401]: why the rename? I don't see how the new name is better than the old.

Comment (hliatis) [reply core/stream/convert.go:401]: I guess because he got rid of dst?

Comment (yalcinmelihyasin) [reply core/stream/convert.go:401]: remove the dst as it is not used.

**Comment by yalcinmelihyasin** [comment]
What is the requested change? I could not understand. Removing the other things except the crash fix?

**Comment by yalcinmelihyasin** [comment]
Also, I have reverted the change in the logic for calculating min/max. Therefore the only change about it is changing the bits. Maybe you have missed that.

**Review by pmuetschard — APPROVED** [review]

**Pull request merged**